File: FinanceApp.Shared/Dto/ForecastParameters/CreateOrUpdateForecastParameters.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Dto.ForecastParameters
{
    public class CreateOrUpdateForecastParameters : UpdateDto
    {
        public double PercentageCdiLoan { get; set; }
        public double PercentageCdiFixedInteresIncometSavings { get; set; }
        public double PercentageCdiVariableIncome { get; set; }
        // Quantos % da entrada de valores no patrimonio deve ser liquido?
        public double SavingsLiquidPercentage { get; set; }
        public int MonthsSavingWarning { get; set; }

    }
}

File: FlutterApp/finance_app/lib/models/forecast_parameters/create_or_update_forecast_parameters.dart
class CreateOrUpdateForecastParameters {
  double percentageCdiLoan;
  double percentageCdiFixedInteresIncometSavings;
  double percentageCdiVariableIncome;
  double savingsLiquidPercentage;
  int monthsSavingWarning;

  CreateOrUpdateForecastParameters({
    required this.percentageCdiLoan,
    required this.percentageCdiFixedInteresIncometSavings,
    required this.percentageCdiVariableIncome,
    required this.savingsLiquidPercentage,
    required this.monthsSavingWarning,
  });

  Map<String, dynamic> toJson() => {
        'percentageCdi': percentageCdiLoan,
        'percentageCdiFixedInteresIncometSavings':
            percentageCdiFixedInteresIncometSavings,
        'percentageCdiVariableIncome': percentageCdiVariableIncome,
        'savingsLiquidPercentage': savingsLiquidPercentage,
        'monthsSavingWarning': monthsSavingWarning
      };
}
